server.js

```diff
@@ -64,5 +81,23 @@ function Weather(jsonObj){
   this.time = jsonObj.valid_date;
 }
 
+
+function Trail(jsonObj){
+  this.name = jsonObj.name;
+  this.location = jsonObj.location;
+  this.length = jsonObj.length;
+  this.stars = jsonObj.stars;
+  this.star_votes = jsonObj.starVotes;
+  this.summary = jsonObj.summary;
+  this.trail_url = jsonObj.url;
+  this.conditions = jsonObj.conditionDetails;
+  this.condition_date = jsonObj.conditionDate.split(' ')[0];
+  this.condition_time = jsonObj.conditionDate.split(' ')[1];
+}
+
+
+// const handleError = (error) => response.status(500).send(error.message);
```

Comment:
I had an issue getting a callback function to work in my error handling. And it looks like you'll run into it here if this is uncommented.
The 'response' variable is not actually being passed to the callback: handleError from the .catch() methods. So you might have to copy your error function into each .catch() method instead.

Reply:
Just a quick note: if you figure out how to pass 'response' to the callback, let me know! It would be so much cleaner if that is possible.